libglusterfs/src/glusterfs/gf-io-common.h

```diff
@@ -92,7 +92,7 @@
 #define gf_res_err(_err)                                                       \
     ({                                                                         \
         int32_t __gf_res_err = -(_err);                                        \
-        if (__gf_res_err < 0) {                                                \
+        if (__gf_res_err > 0) {                                                \
```

Comment:
Wow! this was antipattern for reviews :-)  No wonder it escaped you too earlier.

Reply:
Yes, it was not obvious... :P